src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -0,0 +1,532 @@
+            references
+            if references is not None
+            else js.references.JaxSimModelReferences.zero(
+                model=model, data=data, velocity_representation=VelRepr.Mixed
+            )
+        )
+
+        with (
+            data.switch_velocity_representation(VelRepr.Mixed),
+            references.switch_velocity_representation(VelRepr.Mixed),
+        ):
+            BW_v_WB = data.base_velocity()
+            W_o_dot_B = BW_v_WB[0:3]
+            W_omega_WB = BW_v_WB[3:6]
+            W_v̇_WB, s̈ = js.ode.system_acceleration(
+                model=model,
+                data=data,
+                joint_forces=references.joint_force_references(model=model),
+                link_external_forces_inertial=references.link_forces(
+                    model=model, data=data
+                ),
+            )
+
+        # Convert the inertial-fixed base acceleration to a body-fixed base acceleration.
+        W_H_B = data.base_transform()
+        W_H_BW = W_H_B.at[0:3, 0:3].set(jnp.eye(3))
+        BW_H_W = math.Transform.inverse(W_H_BW)
+        BW_X_W = math.Adjoint.from_transform(
+            BW_H_W,
+        )
+        term1 = BW_X_W @ W_v̇_WB
+        term2 = jnp.zeros(6).at[0:3].set(jnp.cross(W_o_dot_B, W_omega_WB))
+        BW_v̇_WB = term1 - term2
+
+        BW_ν̇ = jnp.hstack([BW_v̇_WB, s̈])
+
+        return BW_ν̇
+
+    @staticmethod
+    def _linear_acceleration_of_collidable_points(
+        model: js.model.JaxSimModel,
+        data: js.data.JaxSimModelData,
+        nu_dot_mixed: jax.Array,
+    ) -> jax.Array:
+        with data.switch_velocity_representation(VelRepr.Mixed):
+            CW_J_WC_BW = js.contact.jacobian(
+                model=model,
+                data=data,
+                output_vel_repr=VelRepr.Mixed,
+            )
+            CW_J_dot_WC_BW = js.contact.jacobian_derivative(
+                model=model,
+                data=data,
+                output_vel_repr=VelRepr.Mixed,
+            )
+
+            BW_ν = data.generalized_velocity()
+
+        CW_a_WC = jax.vmap(
+            lambda J_dot, J, nu_dot, nu: J_dot @ nu + J @ nu_dot,
+            in_axes=(0, 0, None, None),
+        )(
+            CW_J_dot_WC_BW,
+            CW_J_WC_BW,
+            nu_dot_mixed,
+            BW_ν,
+        )
+        print(f"{CW_a_WC.shape=}")
+
+        return CW_a_WC[:, 0:3].squeeze()
+
+    @staticmethod
+    def _compute_baumgarte_stabilization_term(
+        inactive_collidable_points: jax.Array,
+        delta: jax.Array,
+        delta_dot: jax.Array,
+        K: jtp.Float,
+        D: jtp.Float,
+    ) -> jtp.Array:
+        def baumgarte_stabilization(
+            inactive: bool,
+            delta: jax.Array,
+            delta_dot: jax.Array,
+            k_baumgarte: float,
+            d_baumgarte: float,
+        ) -> jtp.Array:
+            baumgarte_term = jax.lax.cond(
+                inactive,
+                lambda in_arg: jnp.zeros(shape=(3)),
```

Comment:
```suggestion
                lambda in_arg: jnp.zeros(shape=(3,)),
```

Reply:
Done in 010ec988da9f7eb5f5fb1863effbcfc6c8f9fb58